### PR TITLE
CVE-2017-12616

### PR DESCRIPTION
**Pull request by allisonsuarez**
History project: curated cves/CVE-2017-12616.yml

### CODE REVIEW

File: cves/CVE-2017-12616.yml [hunk 1]
```diff
@@ -65,7 +69,7 @@ fixes:
      note:
 vccs:
   - commit:
-    note:
+    note: couldn't find github mirror for tomcat70 where the vulnerability is.
```
Comment (mesternefeld) [inline cves/CVE-2017-12616.yml:72]: Take a look here: https://github.com/apache/tomcat/tree/7.0.x

File: cves/CVE-2017-12616.yml [hunk 2]
```diff
@@ -43,7 +43,11 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  The use of the class VirtualDirContext, which is intended to expose
+  multiple directories of the filesystem under a single webapp, made it possible for
+  an attacker to insert a specially crafted request that would enable them to bypass
+  security constraints and view the source code for JavaServer Pages.
```
Comment (Pearse4004) [inline cves/CVE-2017-12616.yml:50]: Any Path traversal vulnerabilities?

File: cves/CVE-2017-12616.yml [hunk 3]
```diff
@@ -4,27 +4,27 @@ nickname_instructions: |
   Nickname is optional. Provide a useful, professional, and catchy nickname for
   this vulnerability. Ideally fewer than 30 characters. This will be shown
   alongside its CVE to make it more easily distinguished from the rest.
-nickname:
+nickname: Dirsclosure
```
Comment (aldenreed) [inline cves/CVE-2017-12616.yml:7]: *Disclosure

File: cves/CVE-2017-12616.yml [hunk 4]
```diff
@@ -202,4 +214,9 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    Not enough testing to be able to catch the unexpected behavior. The fix
+    is just to upgrade to the latest release, which since the vulnerability
+    patch requires at most a minor release upgrade seems like a reasonable
+    solution. The thing that made this vulnerability particularly bad is that
```
Comment (aldenreed) [inline cves/CVE-2017-12616.yml]: This sentence reads kinds of awkwardly to me. I would put the "seems like a reasonable solution" after the word "which".

File: cves/CVE-2017-12616.yml [hunk 5]
```diff
@@ -171,8 +181,10 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: |
+      should have used whitelisting for request inputs, that way the requests
+      used for the vulnerability would have been rejected.
   security_by_obscurity:
```
Comment (aldenreed) [inline cves/CVE-2017-12616.yml:188]: As far as lessons go, could the specially crafted requests be an example of complex inputs?

**Comment by aldenreed** [comment]
I'll give this vulnerability 6 upvotes.

File: cves/CVE-2017-12616.yml [hunk 6]
```diff
@@ -4,27 +4,27 @@ nickname_instructions: |
...
   alongside its CVE to make it more easily distinguished from the rest.
-nickname:
+nickname: Dirsclosure
 
 CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE:
+CWE: CWE-200
```
Comment (Nitepone) [inline cves/CVE-2017-12616.yml]: I don't believe this is correct format. Should just be a number.

File: cves/CVE-2017-12616.yml (hunk 3 above)
Comment (allisonsuarez) [reply cves/CVE-2017-12616.yml:7]: its a made up nickname that combines the word directory and disclosure (as in information disclosure)

Comment (aldenreed) [reply cves/CVE-2017-12616.yml:7]: Oh, sorry. Thought it was a typo.

Comment (andymeneely) [inline cves/CVE-2017-12616.yml:7]: I like it :)

**Pull request merged**